### PR TITLE
MNTOR-3272: API to call FxA apply coupon

### PR DESCRIPTION
**Pull request by mansaj**
# References:

Jira: MNTOR-3272

# Description
- service-side function to apply coupon code for the cancellation flow
- a new ENV VAR to set the coupon code ID

# How to test
In Heroku, subscribe to monthly, make sure the user is subscribed.
Go to settings page and hit cancel

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/utils/fxa.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ async function deleteSubscription(bearerToken) {
         }
       })
       if (!response.ok) {
-        // throw new InternalServerError(`bad response: ${response.status}`)
+        // throw new InternalServerErrorg(`bad response: ${response.status}`)
```
Comment (rhelmer) [inline src/utils/fxa.js:132]: Is this meant to be commented out? Seems to have a typo too ("Errorg")

File: src/app/functions/server/applyCoupon.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { applyCoupon } from "../../../utils/fxa";
+
+export async function applyCurrentCouponCode(
+  subscriber: SubscriberRow | SerializedSubscriber,
+) {
+  logger.info("fxa_apply_coupon_code", {
+    subscriber: subscriber.id,
+  });
+
+  // current coupon
```
Comment (rhelmer) [inline src/app/functions/server/applyCoupon.ts:21]: This comment seems redundant

File: src/app/functions/server/applyCoupon.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+export async function checkCurrentCouponCode(
+  subscriber: SubscriberRow | SerializedSubscriber,
+) {
+  logger.info("fxa_check_coupon", {
+    subscriber: subscriber.id,
+  });
+
+  // current coupon
```
Comment (rhelmer) [inline src/app/functions/server/applyCoupon.ts:68]: Same as above re: comment redundancy

**Review by rhelmer — APPROVED** [review]
lgtm overall, just a few comment-related nits

File: src/utils/fxa.js (hunk 1 above)
Comment (mansaj) [reply src/utils/fxa.js:132]: Not sure why it's commented out, but I adjusted it so it throws and should be caught by the `catch` below

**Pull request merged**